src/plugins/data_source/server/client/client_pool.ts

```diff
@@ -50,12 +56,34 @@ export class OpenSearchClientPool {
     });
     this.logger.info(`Created data source client pool of size ${size}`);
 
-    const getClientFromPool = (endpoint: string) => {
-      return this.cache!.get(endpoint);
+    // aws client specific pool
+    this.awsClientCache = new LRUCache({
+      max: size,
+      maxAge: MAX_AGE,
+
+      async dispose(key, client) {
+        try {
+          await client.close();
+        } catch (error: any) {
+          logger.warn(
+            `Error closing OpenSearch client when removing from aws client pool: ${error.message}`
```

Comment:
Is it a common practice in OSD to issue warnings with the term "Error"? If not, we should change this to
```suggestion
          logger.warn(
            `OpenSearch client failed to close while disposing from the client pool: ${error.message}`
```

Also, why "AWS", that too lowercase?